spanner/client.go

```diff
@@ -456,6 +456,47 @@ func (c *Client) ReadWriteTransaction(ctx context.Context, f func(context.Contex
 	return ts, err
 }
 
+// BeginReadWriteTransaction starts a read-write transaction. Commit() or
+// Rollback() must be called to end a transaction. If Commit() or Rollback() is
+// not called, the session that is used by the transaction will not be returned
+// to the pool and cause a session leak.
+//
+// This method should only be used when manual error handling and retry
+// management is needed. Cloud Spanner may abort a read/write transaction at any
+// moment, and each statement that is executed on the transaction should be
+// checked for an Aborted error, including queries and read operations.
+//
+// For most use cases, client.ReadWriteTransaction should be used, as it will
+// handle all Aborted and 'Session not found' errors automatically.
+func (c *Client) BeginReadWriteTransaction(ctx context.Context) (*ReadWriteTransactionStmtBased, error) {
```

Comment:
Could we move this method to the `ReadWriteTransactionStmtBased` struct, and add a `NewStatementBasedTransaction` method to `client.go` that returns a `ReadWriteTransactionStmtBased`? This method will otherwise stand out a lot to (new) users, and we only want this to be used for advanced purposes and/or library developers.

Reply:
I move the method to `transaction.go` and rename it to `NewReadWriteStmtBasedTransaction`. I don't see a point to keep it in `client.go` any more.